execute/observation.go

```diff
@@ -411,21 +409,24 @@ func (p *Plugin) getFilterObservation(
 				continue
 			}
 
-			nonceRequestArgs[commitReport.SourceChain][sender] = struct{}{}
+			commitReportSenders[commitReport.SourceChain][sender] = struct{}{}
```

Comment:
It looks like you can skip the 2nd loop by changing the type of `commitReportSenders` and appending the sender addresses here.